### PR TITLE
Improve comments on `TyCtxt` and `GlobalCtxt`.

### PR DESCRIPTION
**Pull request by nnethercote**
By adding some non-obvious information that took me a little while to figure out.

r? @compiler-errors

### CODE REVIEW

File: compiler/rustc_middle/src/ty/context.rs [hunk 1]
```diff
@@ -472,12 +472,7 @@ impl<'tcx> TyCtxtFeed<'tcx, LocalDefId> {
     }
 }
 
-/// The central data structure of the compiler. It stores references
```
Comment (compiler-errors) [inline compiler/rustc_middle/src/ty/context.rs:475]: Hm, I feel like `GlobalContext` is a bit of an implementation detail unless you're working on the query system.

At the least, this should link via a doc comment to gcx so people can click to see the more useful doc comment on that instead. Ideally, though, this should explain that the arenas and queries are accessible via tcx as well.

Comment (nnethercote) [reply compiler/rustc_middle/src/ty/context.rs]: I moved the comment back to `TyCtxt` and tweaked it accordingly.

**Review by compiler-errors — APPROVED** [review]

**Pull request merged**